Headlines/src/Controllers/ProfileViewController.swift

```diff
@@ -59,97 +63,153 @@ class ProfileViewController: UITableViewController, TabbedViewController {
-        let r = reactions[indexPath.row]
-        guard let n = r.news else {
-            return cell
+        let fail: ((NSError) -> Void) = { [unowned self] (error) in
+            self.showControllerWithError(error as NSError)
         }
         
-        cell.emojiLabel.text = r.reaction
-        cell.reactionLabel.text = n.title
-        cell.reactionImageView.sd_setImage(with: n.imageUrl)
-        
-        return cell
+        _ = newsService.searchNews(interest.name, success: success, fail: fail)
+    }
+    
+    // MARK: Public
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupCollectionView()
+        fetchProfileData()
     }
     
-    // MARK: UITableViewDelegate
-    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        let r = reactions[indexPath.row]
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
         
-        guard let n = r.news else {
+        guard let identifier = segue.identifier else {
             return
         }
         
-        trackOpenNews(n)
+        switch identifier {
+        case "interest_did_select":
+            guard
+                let vc = segue.destination as? NewsTableViewController,
+                let interestNews = sender as? InterestNews else {
+                return
+            }
+            vc.title = interestNews.interest
+            vc.news = interestNews.news
+            
+        default:
+            break
+        }
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
         
-        openURL(n.url)
+        Answers.logCustomEvent(withName: "profile_appear", customAttributes: nil)
+    }
+    
+    // MARK: UICollectionViewDelegate
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        switch indexPath.section {
+        
+        case 0:
+            let interest = profileDataSource.interests[indexPath.row]
+            handleInterestSelection(interest)
+            break
+        
+        case 1:
+            let reaction = profileDataSource.reactions[indexPath.row]
+            handleReactionSelection(reaction)
+            
+        default:
+            break
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView,
+                        layout collectionViewLayout: UICollectionViewLayout,
+                        sizeForItemAt indexPath: IndexPath) -> CGSize {
+
+        switch indexPath.section {
+        case 1:
+            let xInset = collectionView.layoutMargins.left
+            let yInset = collectionView.layoutMargins.top
+            let cellWidth = collectionView.bounds.insetBy(dx: xInset, dy: yInset).width
+            return CGSize(width: cellWidth, height: 60)
+
+        default:
+            guard let cell = profileDataSource.collectionView(collectionView, cellForItemAt: indexPath) as? LabelCollectionViewCell else {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Canillitapp/headlines-iOS/blob/c203825f89bc24c6a01ab247938473532eae0d63/Headlines/src/Controllers/ProfileViewController.swift#L189"></span>Line should be 120 characters or less: currently 138 characters</td>
    </tr>
  </tbody>
</table>